.github/scripts/build_aosp.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+export DEBIAN_FRONTEND=noninteractive
+
+apt-get update
+apt-get install -y git-core gnupg flex bison build-essential zip curl zlib1g-dev  \
+    libc6-dev-i386 libncurses5 x11proto-core-dev libx11-dev lib32z1-dev libgl1-mesa-dev  \
+    libxml2-utils xsltproc unzip fontconfig python3 npm pip e2fsprogs python3-protobuf \
+    fonts-dejavu diffutils rsync ccache
+
+npm install --global yarn
+
+mkdir -p ~/bin
+curl https://storage.googleapis.com/git-repo-downloads/repo > ~/bin/repo
+chmod a+x ~/bin/repo
+ln -s /usr/bin/python3 /usr/bin/python
+export PATH=~/bin:$PATH
+
+set -e
+
+retry() {
+  set +e
+  local max_attempts=${ATTEMPTS-5}
+  local timeout=${TIMEOUT-1}
+  local attempt=0
+  local exitCode=0
+
+  while [[ $attempt < $max_attempts ]]
+  do
+    "$@"
+    exitCode=$?
+
+    if [[ $exitCode == 0 ]]
+    then
+      break
+    fi
+
+    echo "Failure! Retrying ($*) in $timeout.."
+    sleep "${timeout}"
+    attempt=$(( attempt + 1 ))
+    timeout=$(( timeout * 2 ))
+  done
+
+  if [[ $exitCode != 0 ]]
+  then
+    echo "Failed too many times! ($*)"
+  fi
+
+  set -e
+
+  return $exitCode
+}
+
+DEVICE=$1
+TARGET=$2
+export DISABLE_ROBO_RUN_TESTS=true
+
+# set git identity
+git config --global user.email "seedvault@nowhere.com"
+git config --global user.name "Seedvault"
+
+retry curl https://storage.googleapis.com/git-repo-downloads/repo > ~/bin/repo
+chmod a+x ~/bin/repo
+
+# create custom manifest with Seedvault included
+git clone https://android.googlesource.com/platform/manifest.git
+mv manifest /aosp-manifest
+cd /aosp-manifest
+seedvault_manifest="<remote name=\"seedvault\" fetch=\"https://github.com/$CIRRUS_REPO_OWNER/\" revision=\"$CIRRUS_CHANGE_IN_REPO\" />\n  <project path=\"external/seedvault\" name=\"$CIRRUS_REPO_NAME\" remote=\"seedvault\" />"
+sed -i "/<manifest>/a ${seedvault_manifest}" "default.xml"
+sed -i 's/fetch=".."/fetch="https:\/\/android.googlesource.com\/"/g' "default.xml"
+git add .
+git commit -m "Add Seedvault"
+
+mkdir -p /aosp
+cd /aosp
+retry yes | repo init -u /aosp-manifest -b main --depth=1
```

Comment:
`main` is wrong - that's the development branch, but given Android is developed privately partially, for frameworks stuff it's a mess.

Should make this customisable, and for now could default to

`android-14.0.0_r1` for seedvault `android14` branch.
`android-13.0.0_r32` for `android13`.


Also note that the tags aren't linear, so we can't always just pick the latest `rX` tag. Both of these tags are in our `gradle/dependencies.gradle` though (and we keep that updated)